File: Task/First-class-functions-Use-numbers-analogously/C-sharp/first-class-functions-use-numbers-analogously.cs
using System;
using System.Linq;

class Program
{
    static void Main(string[] args)
    {
        double x, xi, y, yi, z, zi;
        x = 2.0;
        xi = 0.5;
        y = 4.0;
        yi = 0.25;
        z = x + y;
        zi = 1.0 / (x + y);

        var numlist = new[] { x, y, z };
        var numlisti = new[] { xi, yi, zi };
        var multiplied = numlist.Zip(numlisti, (n1, n2) =>
                       {
                           Func<double, double> multiplier = m => n1 * n2 * m;
                           return multiplier;
                       });

        foreach (var multiplier in multiplied)
            Console.WriteLine(multiplier(0.5));
    }
}

File: Task/First-class-functions-Use-numbers-analogously/PascalABC.NET/first-class-functions-use-numbers-analogously.pas
begin
  var x, xi, y, yi, z, zi: real;
  x := 2.0;
  xi := 0.5;
  y := 4.0;
  yi := 0.25;
  z := x + y;
  zi := 1.0 / (x + y);

  var numlist := new real[] ( x, y, z );
  var numlisti := new real[] ( xi, yi, zi );
  var multiplied := numlist.Zip(numlisti, (n1, n2) -> begin
    var multiplier: real -> real := m -> n1 * n2 * m;
    Result := multiplier;
  end);

  foreach var multiplier in multiplied do
    Println(multiplier(0.5));
end.
